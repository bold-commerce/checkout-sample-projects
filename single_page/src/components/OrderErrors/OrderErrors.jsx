import { Message } from '@boldcommerce/stacks-ui';
import React, { useEffect } from 'react';
import { useErrors } from '@boldcommerce/checkout-react-components';

const OrderErrors = () => {
  const { data } = useErrors();

  // Scroll to top of the page if a new error happened
  useEffect(() => {
    window.scrollTo({top: 0, behavior: 'smooth'});
  }, [data?.order]);

  if (data.order && data.order.length > 0) {
    return (
      <Message type="alert">{ data.order[0].message }</Message>
    );
  } else {
    return null;
  }
};

export default OrderErrors;
