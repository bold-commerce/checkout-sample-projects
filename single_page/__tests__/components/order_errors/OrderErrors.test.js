import React from "react";
import { render } from "@testing-library/react";
import OrderErrors from "../../../client/components/OrderErrors/OrderErrors";

jest.mock('@boldcommerce/checkout-react-components', () => ({
  useErrors: () => ({
    data: {
      order: [{ message: 'Testing order errors' }]
    }
  })
}));

describe('OrderErrors', () => {
  test('renders OrderErrors component', () => {
    const { asFragment } = render ( <OrderErrors /> )

    expect(asFragment()).toMatchSnapshot();
  })
})
