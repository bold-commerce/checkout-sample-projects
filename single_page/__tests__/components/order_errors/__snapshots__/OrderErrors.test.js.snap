// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderErrors renders OrderErrors component 1`] = `
<DocumentFragment>
  <div
    class="stx-message stx-message--with-alert"
    role="alert"
  >
    Testing order errors
  </div>
</DocumentFragment>
`;
